client/src/components/products/ProductDetails.js

```diff
@@ -1,51 +1,101 @@
-import React, { useState } from "react";
+import React, { useState, useContext } from "react";
 import Navbar from "../home/Navbar";
 import styled from "styled-components";
-function ProductDetails({ userData, product }) {
-  const [count, setCount] = useState(0);
+import Footer from "../home/Footer";
+import FeaturedProducts from "../home/FeaturedProducts";
+import { CartContext, ProductsContext } from "../context/Context";
+import { useParams } from "react-router";
+function ProductDetails({ userData }) {
+  const [productNo, setProductNo] = useState(1);
+  const [cart, setCart] = useContext(CartContext);
+  const productsData = useContext(ProductsContext);
+  const { id } = useParams();
+  const [product] = productsData.filter((item) => {
+    if (item.id == id) {
+      return item;
+    }
+  });
+  let renderLoading;
+  !product ? (renderLoading = true) : false;
   return (
     <>
       <Navbar userData={userData} />
       <ProductContainer>
-        <div className="product">
-          <img src={product.imageUrl} alt="" />
-        </div>
-        <div className="content">
-          <h2>{product.name}</h2>
-          <div className="price">${product.price}</div>
-          <div className="story">
-            <p>
-              Lorem ipsum dolor sit amet consectetur adipisicing elit. Beatae
-              odit placeat fuga.
-            </p>
-            <p>
-              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Libero
-              voluptatibus expedita omnis molestias magnam molestiae quia
-              blanditiis, illum incidunt impedit.
-            </p>
-            <div className="buttons">
-              <button onClick={() => count && setCount(count - 1)}>-</button>
-              <input
-                type="number"
-                name=""
-                value={count}
-                onChange={(e) => {
-                  setCount(e.target.value);
-                }}
-              />
-              <button onClick={() => setCount(count + 1)}>+</button>
-              <button className="add-cart">Add to Cart</button>
+        {renderLoading ? (
```

Comment:
Why not just put `!product` here instead of `renderLoading`?

Reply:
Sometimes I think of the hardest solution first🤦‍♂️
Thanks!